File: src/main/java/bad/robot/radiate/teamcity/TeamCityEndpoints.java
package bad.robot.radiate.teamcity;

import bad.robot.radiate.Hypermedia;

@Deprecated
public class TeamCityEndpoints {

    public static Hypermedia projectsEndpointFor(final Authorisation authorisation) {
        return () -> String.format("/%s/app/rest/projects", authorisation);
    }

    public static Hypermedia buildsEndpointFor(final Authorisation authorisation) {
        return () -> String.format("/%s/app/rest/builds/", authorisation);
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/TeamCityEndpoints.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.Hypermedia
import bad.robot.radiate.FunctionInterfaceOps.toHypermedia
import bad.robot.radiate.config.Authorisation

object TeamCityEndpoints {
  def projectsEndpointFor(authorisation: Authorisation): Hypermedia = s"/$authorisation/app/rest/projects"
  def buildsEndpointFor(authorisation: Authorisation): Hypermedia = s"/$authorisation/app/rest/builds/"
}
